lib/explainer/explained_query.rb

```diff
@@ -0,0 +1,43 @@
+module Explainer
+  class ExplainedQuery
+    FIELDS = %i(id select_type table partitions type possible_keys key key_len ref rows filtered extra).freeze
+    NULL = 'NULL'.freeze
+
+    FIELDS.each do |field|
+      define_method(field) do
+        explained_query[field]
+      end
+    end
+
+    def initialize(explained_query)
+      validate!(explained_query)
+      @explained_query = explained_query
+    end
+
+    def indexed?
+      type == 'ref' && key != NULL
```

Comment:
We'll likely have to add a bunch of logic like this (incl. probably replacing this) to this class to help us build tests.